tracer/src/Datadog.Trace/IAST/Dataflow/AspectAttribute.cs

```diff
@@ -0,0 +1,157 @@
+    {
+        private static Regex nameSplitter = new Regex(@"(?:([^|]+)\|)?(([^:]+)(?:::[^()]+\(.*\))?)", RegexOptions.Compiled); // 1->Assembly 2->Function 3->Type
+        private readonly List<object> parameters = new List<object>();
+
+        public AspectAttribute(string targetMethod, AspectType aspectType = AspectType.DEFAULT, params VulnerabilityType[] vulnerabilityTypes)
+            : this(targetMethod, string.Empty, 0, false, aspectType, vulnerabilityTypes)
+        {
+        }
+
+        public AspectAttribute(string targetMethod, params AspectFilter[] filters)
+            : this(targetMethod, string.Empty, 0, false, filters)
+        {
+        }
+
+        public AspectAttribute(string targetMethod, string targetType)
+            : this(targetMethod, targetType, 0, false)
+        {
+        }
+
+        public AspectAttribute(string targetMethod, string targetType, int paramShift, bool boxParam, AspectType aspectType = AspectType.PROPAGATION, params VulnerabilityType[] vulnerabilityTypes)
+            : this(targetMethod, targetType, new int[] { paramShift }, new bool[] { boxParam }, new AspectFilter[] { }, aspectType, vulnerabilityTypes)
+        {
+        }
+
+        public AspectAttribute(string targetMethod, string targetType, int paramShift, bool boxParam, AspectFilter[] filters, AspectType aspectType = AspectType.PROPAGATION, params VulnerabilityType[] vulnerabilityTypes)
+            : this(targetMethod, targetType, new int[] { paramShift }, new bool[] { boxParam }, filters, aspectType, vulnerabilityTypes)
+        {
+        }
+
+        public AspectAttribute(string targetMethod, string targetType, int[] paramShift, bool[] boxParam, AspectFilter[] filters, AspectType aspectType = AspectType.PROPAGATION, params VulnerabilityType[] vulnerabilityTypes)
+        {
+            if (paramShift == null || paramShift.Length == 0) { paramShift = new int[] { 0 }; }
+            if (boxParam == null || boxParam.Length == 0) { boxParam = new bool[] { false }; }
+            if (filters == null || filters.Length == 0) { filters = new AspectFilter[] { AspectFilter.None }; }
+
+            if (paramShift.Length > 1 && boxParam.Length == 1)
+            {
+                boxParam = Enumerable.Repeat(boxParam[0], paramShift.Length).ToArray();
+            }
+
+            Debug.Assert(paramShift.Length == boxParam.Length, "paramShift and boxParam must be same len");
+
+            parameters.Add(targetMethod.Quote());
+            parameters.Add(targetType.Quote());
+            parameters.Add(paramShift);
+            parameters.Add(boxParam);
+            parameters.Add(filters);
+            parameters.Add(aspectType);
+            parameters.Add(string.Join(",", (vulnerabilityTypes ?? new VulnerabilityType[0]).Select(i => i.ToString()).ToArray()).Quote());
+
+            var targetMethodMatch = nameSplitter.Match(targetMethod);
+            TargetMethodAssemblies = GetAssemblyList(targetMethodMatch.Groups[1].Value);
```

Comment:
what happens if the match fails, will it give a compile error as the attribute can't be generated? If so, is the message meaningful?

Reply:
Uhm, I have to check that